src/Compilers/VisualBasic/Portable/Scanner/Directives.vb

```diff
@@ -50,10 +50,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 Me.GetNextTokenInState(ScannerState.VB)
 
             Else
-                Dim parser As New Parser(Me)
-
-                directiveTrivia = parser.ParseConditionalCompilationStatement()
-                directiveTrivia = parser.ConsumeStatementTerminatorAfterDirective(directiveTrivia)
+                Using parser As New Parser(Me)
```

Comment:
`Scanner.Dispose` is not called when `Parser` is instantiated through this constructor, so no double dispose here. However, I agree a better design is to have an explicit ctor parameter, rather than tight coupling on current implementation.